src/structures/GuildMember.js

```diff
@@ -71,18 +64,27 @@ class GuildMember extends Base {
   }
 
   _patch(data) {
-    /**
-     * The nickname of this member, if they have one
-     * @type {?string}
-     * @name GuildMember#nickname
-     */
-    if (typeof data.nick !== 'undefined') this.nickname = data.nick;
+    if ('user' in data) {
+      /**
+       * The user that this guild member instance represents
+       * @type {User}
+       * @name GuildMember#user
+       */
+      this.user = this.client.users.add(data.user, true);
+    }
 
-    if (data.joined_at) this.joinedTimestamp = new Date(data.joined_at).getTime();
-    if (data.premium_since) this.premiumSinceTimestamp = new Date(data.premium_since).getTime();
+    if ('nick' in data) {
+      /**
+       * The nickname of this member, if they have one
+       * @type {?string}
+       * @name GuildMember#nickname
+       */
+      this.nickname = data.nick;
```

Comment:
While you are at it, can you update the typings for `nickname`?
Its currently nullable while it should be `nickname?: string;`